admin-auth/src/main/kotlin/com/epam/drill/admin/auth/service/impl/OAuthServiceImpl.kt

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2020 - 2022 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.epam.drill.admin.auth.service.impl
+
+import com.epam.drill.admin.auth.config.OAuthConfig
+import com.epam.drill.admin.auth.config.OAuthUnauthorizedException
+import com.epam.drill.admin.auth.entity.UserEntity
+import com.epam.drill.admin.auth.model.UserInfoView
+import com.epam.drill.admin.auth.principal.Role
+import com.epam.drill.admin.auth.repository.UserRepository
+import com.epam.drill.admin.auth.service.OAuthService
+import io.ktor.auth.*
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import kotlinx.serialization.json.*
+
+class OAuthServiceImpl(
+    private val httpClient: HttpClient,
+    private val oauthConfig: OAuthConfig,
+    private val userRepository: UserRepository
+) : OAuthService {
+
+    override suspend fun signInThroughOAuth(principal: OAuthAccessTokenResponse.OAuth2): UserInfoView {
+        val oauthUser = getUserInfo(principal.accessToken).toEntity()
+        val dbUser = userRepository.findByUsername(oauthUser.username)
+        if (dbUser?.blocked == true)
+            throw OAuthUnauthorizedException("User is blocked")
```

Comment:
> We are still in the process of OAuth2 authenticating there
Yes, but my concern is that it'll look confusing outside of this function (e.g. in log). "Hey look, `OAuthUnauthorizedException` happened, because user is blocked" - there is ambiguity in _where_ user is blocked, and it will require digging in Drill4J code to find out where the issue is.

We can keep the exception type, just make it clear (somehow) that user is blocked in Drill4J db and not by OAuth provider 

> ... and remove this message for user
_tbd_ how

Reply:
I guess dedicated exception type resolves the issue + we can document that
https://github.com/Drill4J/admin/pull/346/commits/839dd6ee1381c2fb11086549da2572fa2d2d0a1e#diff-588f6a4c58b26ece3d7d18ca210f0598213569fc05c7f673f62e8b073bf41cd5R118